tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+import time
+
+from . import common
+
+from odoo.exceptions import ValidationError
+
+
+ODOO_DATETIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+class MockPIOutcome:
+    def __init__(self, got_changes):
+        self._got_changes = got_changes
+
+    def got_inventory_changes(self):
+        return self._got_changes
+
+
+class TestLocationPI(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestLocationPI, cls).setUpClass()
+        Location = cls.env['stock.location']
+        User = cls.env['res.users']
+
+        user_warehouse = User.get_user_warehouse()
+        cls.picking_type_pick = user_warehouse.pick_type_id
+        cls.datetime_tolerance = timedelta(minutes=1)
+
+        loc_id = 9834247
+
+        while Location.browse(loc_id).exists():
+            if loc_id > 9835147:
+                # done 100 attempts, that's enough
+                cls.fail("Failed to determine an unknown location; "
+                         "please use an empty database for testing")
+            else:
+                loc_id += 9
+
+        cls.unknown_location_id = loc_id
+
+    def setUp(self):
+        super(TestLocationPI, self).setUp()
+        Package = self.env['stock.quant.package']
+
+        self.package_one = Package.get_package("test_pi_package_one",
+                                               create=True)
+        self.package_two = Package.get_package("test_pi_package_two",
+                                               create=True)
+
+    def test01_validate_pi_request_success(self):
+        """ No error is raised for a good PI request """
```

Comment:
`when validating a good PI request` instead of `for ...` ?

Do we have something similar for a request with inventory adjustments?

Reply:
hmmm, `when validating` is superfluous, as you have the method name in the test name...
I prefer to reduce code / comment bloat, but will improve the docstring.

Reply:
Will add an analogous for a an inventory adjustment.